[Term Entry] Cybersecurity Malware keylogger #2689 (#2728)
Codecademy/docs: content/cybersecurity/concepts/malware/terms/keylogger/keylogger.md

---
Title: 'Keylogger'
Description: 'Keyloggers are a type of computer malware that records keystroke events on the keyboard and saves them to a log file, allowing it to steal sensitive data like passwords.'
Subjects:
  - 'Cybersecurity'
  - 'Computer Science'
Tags:
  - 'Cybersecurity'
  - 'Malware'
CatalogContent:
  - 'introduction-to-cybersecurity'
  - 'paths/fundamentals-of-cybersecurity'
---

> **Keyloggers** are a type of computer [Malware](https://www.codecademy.com/resources/docs/cybersecurity/malware) that records keystroke events on the keyboard and saves them to a log file, allowing it to steal sensitive data.

Reviewer: ```suggestion
**Keyloggers** are a type of computer [Malware](https://www.codecademy.com/resources/docs/cybersecurity/malware) that records keystroke events on the keyboard and saves them to a log file, allowing them to steal sensitive data.
```

> >*Keystroke logging is one of the methods to obtain sensitive informations like usernames, PINs, password and online activities from endpoint devices. An attacker can collect valuable data without entering into a strong database or file server using this method.*

Reviewer: ```suggestion
Keystroke logging is one of the methods to obtain sensitive information like usernames, PINs, passwords, and online activities from endpoint devices. An attacker can collect valuable data without entering into a strong database or file server using this method.
```

## How Keyloggers Work

> Key loggers, also known as keystroke loggers, record the keys hit on adevice and save them to a file, which is then accessed by the person who deployed the malware.

Reviewer: ```suggestion
Keyloggers, also known as keystroke loggers, record the keys hit on a device and save them to a file, which is then accessed by the person who deployed the malware.
```

> The main purpose of keyloggers is to tamper with the chain of events that occur when a key is pressed, and information is displayed on the screen as a result of the keystroke.

Reviewer: ```suggestion
The main purpose of keyloggers is to tamper with the chain of events that occur when a key is pressed and information is displayed on the screen as a result of the keystroke.
```

> Keyloggers can be used for both lawful and illegitimate objectives,depending on the user who is utilising it.

Reviewer: ```suggestion
Keyloggers can be used for both lawful and illegitimate objectives, depending on the user who is utilizing it.
```

> ***Lawful Application include:***

Reviewer: ```suggestion

Lawful Application include:
```

> > - *Developers and analysts studying user       interaction with systems*.
> > - *Employee monitoring.*
> > - *Law enforcement or private investigators looking for evidence of an ongoing crime or inappropriate behavior.*
> > - *Quality assurance testers analyzing sources of system errors .*

Reviewer: ```suggestion
- Developers and analysts studying user interaction with systems.
- Employee monitoring.
- Law enforcement or private investigators looking for evidence of an ongoing crime or inappropriate behavior.
- Quality assurance testers analyzing sources of system errors .
```

> On the other side of the line between lawful and unlawful use, cybercriminals use keylogging technology to capture identities, confidential intellectual property, passwords and any other marketable information.

Reviewer: ```suggestion
On the other hand, unlawful use consists of cybercriminals using keylogging technology to capture identities, confidential intellectual property, passwords and any other marketable information.
```

> ### Types of Keyloggers

Reviewer: ```suggestion
## Types of Keyloggers
```

> ***Keyloggers*** fall into four categories: software, hardware, wireless intercept, and acoustic.Although they differ in how they are implemented and how information is captured,these keyloggers have one thing in common.They store capture information in a log file.

Reviewer: ```suggestion
Keyloggers fall into four categories: software, hardware, wireless intercept, and acoustic. Although they differ in how they are implemented and how information is captured, these keyloggers have one thing in common. They store capture information in a log file.
```

> >"*When software or hardware keyloggers are used, the log files are stored on the compromised machine.

Reviewer: ```suggestion
When software or hardware keyloggers are used, the log files are stored on the compromised machine.
```

> Remote capture technologies (i.e., wireless intercept and acoustic) typically store keystroke data on the collection device*".

Reviewer: ```suggestion
Remote capture technologies (i.e., wireless intercept and acoustic) typically store keystroke data on the collection device.
```

> #### 1.*Software Keyloggers*

Reviewer: ```suggestion
### Software Keyloggers
```

> Software keyloggers capture keystroke information as it passes between the computer keyboard interface and the OS. They are implemented as traditional applications or kernel-based. In almost all malicious instances of this type of keylogger, users participated in some way in the software’s installation.

Reviewer: ```suggestion
Software keyloggers capture keystroke information as it passes between the computer keyboard interface and the OS. They are implemented as traditional applications or are kernel-based. In almost all malicious instances of this type of keylogger, users participated in some way in the software’s installation.
```

> #### 2.*Hardware Keyloggers*

Reviewer: ```suggestion
### Hardware Keyloggers
```

A hardware keylogger is essentially a circuit located somewhere between the keyboard and the computer. The keylogger is connected directly to the PC and the keyboard to the keylogger.
Another method is to install a keylogger circuit into a standard keyboard.This has the advantage of no physical evidence of user monitoring.

> #### 3.*wireless intercept Keyloggers*

Reviewer: ```suggestion
### Wireless Intercept Keyloggers
```

> A wireless intercept keylogger usees wireless Technologies based on the capabality of transmission.For instance Bluetooth-accessible keylogger.

Reviewer: ```suggestion
A wireless intercept keylogger uses wireless technologies based on the capability of transmission. For instance a Bluetooth-accessible keylogger.
```

> The one big disadvantage of using wireless intercept keyloggers is the need for a receiver/antenna relatively close to the target system because the wireless devices are limited range of transmission.

Reviewer: ```suggestion
The one big disadvantage of using wireless intercept keyloggers is the need for a receiver/antenna relatively close to the target system because the wireless devices have a limited range of transmission.
```

> #### 4.*Acoustic Keyloggers*

Reviewer: ```suggestion
### Acoustic Keyloggers
```

A acoustic keylogger requires special equipment that “listens” to a user typing and special software that performs statistical analysis on captured data.
The devices used to remotely listen to conversations are used to record typing sounds. Such microphones can be placed in the target work area or long distance solutions can be used.

> >*Parabolic microphones are an example of a long distance device.These microphones can pick up keyboard sounds from hundreds of feet away. Attached equipment records the sounds, which are then passed to audio-to-character translation software.*

Reviewer: ```suggestion
Parabolic microphones are an example of a long distance device. These microphones can pick up keyboard sounds from hundreds of feet away. Attached equipment records the sounds, which are then passed to audio-to-character translation software.
```
